### PR TITLE
Fix gss.Rmd to use params$ to access parameterized values. Add some i…

### PR DESCRIPTION
**Pull request by blankenberg**
…nput path and output file options. Some lintr.

### CODE REVIEW

**Review by shbrief — COMMENTED** [review]
For `gss.Rmd`, is there any reason you use `params$` instead of defining them in the `params` field within the YAML header?

**Comment by blankenberg** [comment]
Hi @shbrief my understanding is that using the params$param_name is the proper way to handle parameterizing the call in gss.Rmd. Some explanation:
1. Call gss.Rmd from gss.R (gss.Rmd actually inherits the environment/workspace including variables from gss.R, which is why it was previously working).
2. I renamed some of the variables in gss.R (to pass lintr linting tests; we are not currently linting *.Rmd files in tools-iuc), this caused gss.Rmd to start failing, because those variables are no longer defined in .R to be automatically made available in .Rmd (I would consider that this worked before to be bad namespacing/scoping, but that is R for you).
3. We are explicitly passing in to `rmarkdown::render` a list of parameters that should be used. To access these within gss.Rmd, we need to pull them out of the `params$` list either by assigning to a 'flat' local name, or just accessing directly from the list as needed.

Basically, I think the reason it was working before is because the call to `rmarkdown::render` inherits the whole current environment/workspace (scary behavior, IMHO), to use it properly as a module that you might call from e.g. another gss_new.R script, we should be using the passed in parameters and not ones that might be matching (accidentally or intentionally) by name. 

More than happy to be wrong here, my R is not nearly as good as my Python.

**Comment by shbrief** [comment]
Got it! I didn't realize that I need to extract `params` - I misunderstood that specifying parameters within the YAML header is all I need to set the new variables within Rmarkdown. Yes, don't want to inherit the whole environment/workspace. Thanks for the help!
